R/steps-rcmdcheck.R

```diff
@@ -7,7 +7,9 @@ TicStepWithPackageDeps <- R6Class(
     prepare = function() {
       verify_install("remotes")
 
-      remotes::install_deps(dependencies = TRUE)
+      repos = c(getOption("repos"), remotes::bioc_install_repos()[1])
```

Comment:
Can you please use `<-` for consistency with the rest of the code? Also, could you please add a comment why you're selecting only the first repo. Perhaps some more detail could be added to `?bioc_install_repos` too.

Reply:
> Also, could you please add a comment why you're selecting only the first repo.

Simply because I do not see the added value of having 3 Bioc repos in the list. One should be sufficient?

Reply:
> Perhaps some more detail could be added to ?bioc_install_repos too.

This fun is from the `remotes` pkg and we call it internally to prepare the step so it is out of ours and the users scope. Or did I misunderstand you here?

Reply:
But these sub-repos all contain different packages?

``` r
library(tidyverse)

pkgs <-
  remotes::bioc_install_repos() %>%
  enframe() %>%
  mutate(available = map(value, ~ rownames(available.packages(repos = .)))) %>%
  select(-value)

repo_pairs <-
  crossing(name = pkgs$name, other_name = pkgs$name) %>%
  left_join(pkgs, by = "name") %>%
  left_join(pkgs, by = c("other_name" = "name")) %>%
  mutate(available_both = map2_int(available.x, available.y, ~ length(intersect(.x, .y))))

repo_pairs %>%
  xtabs(formula = available_both ~ name + other_name)
#>           other_name
#> name       BioCann BioCexp BioCsoft
#>   BioCann      922       0        1
#>   BioCexp        0     342        1
#>   BioCsoft       1       1     1562
```

<sup>Created on 2018-11-15 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1.9000)</sup>

It's in the scope of whoever reads the _tic_ code. Either we comment it here, or we add a comment to the function in _remotes_, or both.

Reply:
> But these sub-repos all contain different packages?

Oh I did not know that - I thought they were just mirrors. I have no experience with Bioc at all and just stumbled upon that problem bc some rev dep of _mlr_ is now a Bioc pkg...
Hope finding that out did not take you too long!

> Either we comment it here, or we add a comment to the function in remotes, or both.

You mean just an info about which repos we use to prepare the check step? Or additional options, like giving the user control over the Bioc versions that will be used? Feel free to tweak the doc to your liking :)

Reply:
Yeah, I stumbled over the `[1]`. Without it, the code and accompanying docs in _remotes_ would seem enough.

Reply:
Using all bioc repos now.
Also I added a short info to `?step_rcmdcheck()`.